tmt/steps/execute/__init__.py

```diff
@@ -55,77 +58,192 @@
+    variable. All scripts must be located in this directory.
 
-    path: Path
-    aliases: list[Path]
+    The default destination directory of the scripts is :py:data:`DEFAULT_SCRIPTS_DEST_DIR`.
+    On ``rpm-ostree`` distributions like Fedora CoreOS, the default destination
+    directory is :py:data:``DEFAULT_SCRIPTS_DEST_DIR_OSTREE``. The destination directory
+    of the scripts can be forced by the script using ``destination_path`` attribute.
+
+    The destination directory can be overridden using the environment variable defined
+    by the :py:data:`DEFAULT_SCRIPTS_DEST_DIR_VARIABLE` variable.
+
+    The ``enabled`` attribute can specify a function which is called with :py:class:`Guest`
+    instance to evaluate if the script is enabled. This can be useful to optionally disable
+    a script for specific guests.
+    """
+
+    source_filename: str
+    destination_path: Optional[Path]
+    aliases: list[str]
     related_variables: list[str]
+    enabled: Callable[[Guest], bool]
+
+    def __enter__(self) -> Path:
+        return SCRIPTS_SRC_DIR / self.source_filename
+
+    def __exit__(self, *args: object) -> None:
+        pass
 
 
 @dataclass
 class ScriptCreatingFile(Script):
-    """ Represents a script which creates a file """
+    """
+    Represents a script which creates a file.
+
+    See :py:class:`Script` for more details.
+    """
 
     created_file: str
 
 
+@dataclass
+class ScriptTemplate(Script):
+    """
+    Represents a Jinja2 templated script.
+
+    Must be used as a context manager. The context manager returns
+    the source file path.
+
+    The source filename is constructed from the name of the file specified
+    via the ``source_filename`` attribute, with the ``.j2`` suffix appended.
+    The template file must be located in the directory specified
+    via :py:data:`SCRIPTS_SRC_DIR` variable.
+    """
+
+    context: dict[str, str]
+
+    _rendered_script_path: Optional[Path] = None
+
+    def __enter__(self) -> Path:
+        with NamedTemporaryFile(mode='w', delete=False) as rendered_script:
+            rendered_script.write(render_template_file(
+                SCRIPTS_SRC_DIR / f"{self.source_filename}.j2", None, **self.context))
+
+        self._rendered_script_path = Path(rendered_script.name)
+
+        return self._rendered_script_path
+
+    def __exit__(self, *args: object) -> None:
+        assert self._rendered_script_path
+        os.unlink(self._rendered_script_path)
+
+
+def effective_scripts_dest_dir(default: Path = DEFAULT_SCRIPTS_DEST_DIR) -> Path:
+    """
+    Find out what the actual scripts destination directory is.
+
+    If the ``TMT_SCRIPTS_DEST_DIR`` environment variable is set, it is used
+    as the scripts destination directory. Otherwise, the ``default``
+    is returned.
+    """
+
+    if SCRIPTS_DEST_DIR_VARIABLE in os.environ:
+        return Path(os.environ[SCRIPTS_DEST_DIR_VARIABLE])
+
+    return default
```

Comment:
can be just
```suggestion
    return Path(os.environ.get(SCRIPTS_DEST_DIR_VARIABLE, default))
```
seems to be working fine even for Path(Path())
but either way is fine with me

Reply:
save 3 lines, good point